### ISSUE TITLE
>1.3.0 Default Fragment overridden if used in th:each loop

### ISSUE THREAD

**Issue by Antibrumm** [ultraq/thymeleaf-layout-dialect#87]
Hi 
I have just tried to update your library from 1.2.9 to 1.3.1 and I ran into an issue while using the "Default" layout feature inside a th:each loop. This was working with 1.2.9. 

When you declare a `layout:fragment` with a default implementation it will be super seeded for all  iterations following a specified part.

I believe that it might have something to do with where the fragment name local variable is stored. 

I was able to reproduce it using these templates:

**Fragment**: defaults to a `span`
```html
<!DOCTYPE html>
<html xmlns:th="http://www.thymeleaf.org"
	xmlns:layout="http://www.ultraq.net.nz/web/thymeleaf/layout">
<body>
	<div layout:fragment="text">
		<label th:text="${text}"></label>
		<div layout:fragment="field" th:remove="tag">
			<span th:text="${text}"></span>
		</div>
	</div>
</body>
</html>
```

**Page**: Use default except for the second element which is an `input`
```html
<!DOCTYPE html>
<html xmlns:th="http://www.thymeleaf.org"
	xmlns:layout="http://www.ultraq.net.nz/web/thymeleaf/layout">
<head></head>
<body>
	<div th:each="a,itr : ${aList}" th:with="text=${a.text}">
		<div th:if="${itr.index != 1}" th:remove="tag">
			<div layout:include="layout/fragment :: text" th:remove="tag">
				<!-- DEFAULT -->
			</div>
		</div>
		<div th:if="${itr.index == 1}" th:remove="tag">
			<div layout:include="layout/fragment :: text" th:remove="tag">
				<div layout:fragment="field">
					<input type="text" th:value="${text}" />
				</div>
			</div>
		</div>
	</div>
</body>
</html>
```

**Result**: All elements after the second are rendered as the last used specific fragment
```html
<!DOCTYPE html>
<html>
<head></head>
<body>
	<div>
		<label>TEST 0</label> <span>TEST 0</span>
	</div>
	<div>
		<label>TEST 1</label> <input type="text" value="TEST 1" />
	</div>
	<div>
		<label>TEST 2</label> <input type="text" value="TEST 2" /> <!-- should be span again -->
	</div>
	<div>
		<label>TEST 3</label> <input type="text" value="TEST 3" /> <!-- should be span again -->
	</div>
</body>
</html>
```

The test was rendered with the help of a simple spring controller and a basic model A:
```java
@Controller
public class LayoutTestController {

    public static class A {
        private String text;

        public String getText() {
            return text;
        }

        public void setText(String text) {
            this.text = text;
        }
    }

    @RequestMapping("test/layoutDialect")
    public String testLayoutDefaults(Model model) {
        List<A> aList = new ArrayList<>();
        for (int i = 0; i < 4; i++) {
            A a = new A();
            a.setText("TEST " + i);
            aList.add(a);
        }
        model.addAttribute("aList", aList);
        return "layout/test";
    }
}
```

**Comment by ultraq** [comment]
Thanks for the bug report - I think I'll be able to convert your example to a test case to fix it.

**Comment by ultraq** [comment]
Fixed!  As you suspected, the place the fragments were stored was the culprit.  In 1.3.0 I introduced a fragment map, but it was a global that then meant each decorator/include/replace operation would share fragments, which isn't desired.

I've created a 1.3.2-SNAPSHOT version with this fix.  This was the last bug fix I planned for 1.3.2 though, so a proper release should be coming soon.
